backend/raspberry_interactions/server_communications.py

```diff
@@ -51,22 +58,36 @@ def terminal_message(self, message):
         if self.on_terminal_msg:
             self.on_terminal_msg(message)
 
+    def register_device(self, topic_type: str, ip: str, registered_count: int, registered_list: List[int]) -> None:
+        topic = f"{topic_type}{registered_count}/"
+
+        self.client.subscribe(topic)
+        self.subscribed_topics.append(topic)
+        self.send_message(GREETING_TOPIC, f"for#{ip}#{registered_count}")
+        registered_list.append(registered_count)
+
+        if topic_type == TERMINAL_TOPIC:
+            self.terminals_products_dict[self.registered_terminals_count] = None
```

Comment:
this line, instead of being inside an IF, should be in the greeting from raspberry part, i feel, you already have all the necessary info there, and you avoid another IF